typegraph/core/src/runtimes/prisma/type_generation/mod.rs

```diff
@@ -0,0 +1,204 @@
+        ))
+    }
+
+    fn find_many_inp(&mut self, model_id: TypeId) -> Result<TypeId> {
+        Ok(t::struct_()
+            .prop(
+                "where",
+                t::optional(self.generate(&QueryWhereExpr::new(model_id))?).build()?,
+            )
+            .prop(
+                "orderBy",
+                t::optional(self.generate(&OrderBy::new(model_id, vec![]))?).build()?,
+            )
+            .prop("take", t::optional(self.generate(&Take)?).build()?)
+            .prop("skip", t::optional(self.generate(&Skip)?).build()?)
+            .prop(
+                "distinct",
+                t::optional(self.generate(&Distinct(model_id))?).build()?,
+            )
+            .build()?)
+    }
+
+    pub fn find_many(&mut self, model_id: TypeId) -> Result<(TypeId, TypeId)> {
+        self.registry.manage(model_id)?;
+
+        Ok((
+            // input
+            self.find_many_inp(model_id)?,
+            // output
+            // t::integer().build()?
+            t::array(self.generate(&WithNestedCount::new(model_id))?).build()?,
+        ))
+    }
+
+    pub fn find_first(&mut self, model_id: TypeId) -> Result<(TypeId, TypeId)> {
+        self.registry.manage(model_id)?;
+
+        Ok((
+            // input
+            self.find_many_inp(model_id)?,
+            // output
+            t::optional(self.generate(&OutType::new(model_id))?).build()?,
+        ))
+    }
+
+    pub fn create_one(&mut self, model_id: TypeId) -> Result<(TypeId, TypeId)> {
+        self.registry.manage(model_id)?;
+
+        Ok((
+            // input
+            t::struct_()
+                .prop("data", self.generate(&InputType::for_create(model_id))?)
+                .build()?,
+            // output
+            self.generate(&OutType::new(model_id))?,
+        ))
+    }
+
+    pub fn create_many(&mut self, model_id: TypeId) -> Result<(TypeId, TypeId)> {
+        self.registry.manage(model_id)?;
+
+        Ok((
+            // input
+            t::struct_()
+                .prop(
+                    "data",
+                    t::array(self.generate(&InputType::for_create(model_id))?).build()?,
+                )
+                .build()?,
+            // output
+            t::struct_().prop("count", t::integer().build()?).build()?,
+        ))
+    }
+
+    pub fn update_one(&mut self, model_id: TypeId) -> Result<(TypeId, TypeId)> {
+        self.registry.manage(model_id)?;
+
+        Ok((
+            // input
+            t::struct_()
+                .prop("data", self.generate(&InputType::for_update(model_id))?)
+                .prop(
+                    "where",
+                    self.generate(&QueryUniqueWhereExpr::new(model_id))?,
+                )
+                .build()?,
+            //output
+            self.generate(&OutType::new(model_id))?,
+        ))
+    }
+
+    fn generate(&mut self, generator: &impl TypeGen) -> Result<TypeId> {
+        let type_name = generator.name(self);
```

Comment:
Maybe add a comment why choose this approach instead of calling `generator.generator(context)` directly ?